internal/gcsx/file_cache_reader_test.go

```diff
@@ -699,3 +700,79 @@ func (t *fileCacheReaderTest) Test_Destroy_NilCacheHandle() {
 
 	assert.Nil(nil, t.reader.fileCacheHandle)
 }
+
+func (t *fileCacheReaderTest) Test_Concurrent_ReadAt() {
```

Comment:
Should we add a timeout logic for this test - high enough, to ensure failure for deadlock scenario.